src/main/java/com/couchbase/lite/internal/Body.java

```diff
@@ -126,8 +124,9 @@ public String getJSONString() {
     public Map<String, Object> getProperties() {
         Object object = getObject();
         if(object instanceof Map) {
-            Map<String, Object> map = (Map<String, Object>) object;
-            return Collections.unmodifiableMap(map);
```

Comment:
@pasin and @tleyden 
Body.getProperties() is called internally.
- Body.getPropertyForKey(String)
- RevisionInternal.getProperties()
- Router.do_PUT_Document(Database, String, String)
I believe `Collections.unmodifiableMap()` is unnecessary.